_posts/2018/2018-01-07-be-sparing-when-using-unless.md

```diff
@@ -0,0 +1,66 @@
+---
+title: 'Be sparing when using unless'
+description: 'Gimme some sugar, but not too much'
+layout: article
+category: ruby
+image:
+  base: '2018/be-sparing-when-using-unless'
+  alt: 'Two directions sign'
+  source: 'https://unsplash.com/photos/lPQIndZz8Mo'
+  credit: 'Pablo Garcia Saldaña'
+---
+
+One of Ruby’s strengths is its delightful, built-in syntactic sugar. One example of this is the keyword `unless`, which you can use in place of using `if` combined with a negative statement.
+
+However, given the flexibility of Ruby's syntax, it is easy to make code harder to understand than it needs to be.
+
+## Instead of…
+
+...using `unless` in ways that makes your brain hurt.
+
+```ruby
+# Example 1
+unless something?
+  # do something
+else
+  # do other thing
+end
+
+# Example 2
+unless something? || another_thing?
+  # do something
+end
+```
+
+
+## Use…
+
+...`if` where it makes the code clearer.
+
+```ruby
+# Example 1
+if !something
+  # do other thing
+else
+  # do something
+end
+
+# Example 2
+if !something? && !another_thing?
+  # do something
+end
+```
+
+
+## But why?
+
+Avoid using an `else` block with an `unless`. This structure is hard to reason about because the `else` block is a double negative: it is "not not" the specified condition. Instead, replace the `unless` with an `if` while refactoring the condition and swap around the blocks of code in the main and `else` branches.
```

Comment:
* Instead, replace the `unless` with an `if`, taking care to refactor the condition statement as is necessary, and swap around...